kahuna/public/js/components/gr-archiver/gr-archiver.js

```diff
@@ -0,0 +1,62 @@
+import angular from 'angular';
+import Rx from 'rx';
+
+import './gr-archiver.css!';
+import template from './gr-archiver.html!text';
+
+import '../../services/archive';
+import '../../services/image-accessor';
+
+
+export const module = angular.module('gr.archiver', [
+    'kahuna.services.archive',
+    'kahuna.services.image-accessor'
+]);
+
+module.controller('grArchiverCtrl', [
+    '$scope',
+    'archiveService',
+    'imageAccessor',
+    function ($scope,
+              archiveService,
+              imageAccessor) {
+
+        const ctrl = this;
+
+        $scope.$watchCollection(() => ctrl.images.toJS(), (images) => {
+            const allArchived = images.every(imageAccessor.isArchived);
+            ctrl.archivedState = allArchived ? 'archived' : 'unarchived';
+        });
+
+        ctrl.archive = () => {
+            ctrl.archiving = true;
+            var imageArray = Array.from(ctrl.images);
```

Comment:
I guess I was coming at it the other way, making the directive compatible with any `Iterable` object, rather than specifically ImmutableJS. So for example if the images end up refactored into a JS array, things will keep working. Probably overly superstitious, I admit.